planning/scenario_planning/common/motion_velocity_optimizer/src/motion_velocity_optimizer.cpp

```diff
@@ -14,177 +14,204 @@
  * limitations under the License.
  */
 
-#include <chrono>
 #include <motion_velocity_optimizer/motion_velocity_optimizer.hpp>
 #include <motion_velocity_optimizer/optimizer/l2_pseudo_jerk_optimizer.hpp>
 #include <motion_velocity_optimizer/optimizer/linf_pseudo_jerk_optimizer.hpp>
 
-// clang-format on
-MotionVelocityOptimizer::MotionVelocityOptimizer() : nh_(""), pnh_("~"), tf_listener_(tf_buffer_)
+#include <tf2_ros/create_timer_ros.h>
+#include <chrono>
+
+using std::placeholders::_1;
+
+#define UPDATE_PARAM(NAME) update_param(parameters, #NAME, planning_param_.NAME);
+namespace
+{
+template <typename T>
+void update_param(
+  const std::vector<rclcpp::Parameter> & parameters, const std::string & name, T & value)
+{
+  auto it = std::find_if(
+    parameters.cbegin(), parameters.cend(),
+    [&name](const rclcpp::Parameter & parameter) { return parameter.get_name() == name; });
+  if (it != parameters.cend()) {
+    value = it->template get_value<T>();
+  }
+}
+}  // namespace
+
+MotionVelocityOptimizer::MotionVelocityOptimizer() : Node("motion_velocity_optimizer")
 {
   auto & p = planning_param_;
-  pnh_.param<double>("max_velocity", p.max_velocity, 20.0);  // 72.0 kmph
-  pnh_.param<double>("max_accel", p.max_accel, 2.0);         // 0.11G
-  pnh_.param<double>("min_decel", p.min_decel, -3.0);        // -0.2G
-
-  pnh_.param<double>("max_lateral_accel", p.max_lateral_accel, 0.2);  //
-  pnh_.param<double>("decel_distance_before_curve", p.decel_distance_before_curve, 3.5);
-  pnh_.param<double>("decel_distance_after_curve", p.decel_distance_after_curve, 0.0);
-  pnh_.param<double>("min_curve_velocity", p.min_curve_velocity, 1.38);
-
-  pnh_.param<double>("replan_vel_deviation", p.replan_vel_deviation, 3.0);
-  pnh_.param<double>("engage_velocity", p.engage_velocity, 0.3);
-  pnh_.param<double>("engage_acceleration", p.engage_acceleration, 0.1);
-  pnh_.param<double>("engage_exit_ratio", p.engage_exit_ratio, 0.5);
+  p.max_velocity = declare_parameter("max_velocity", 20.0);  // 72.0 kmph
+  p.max_accel = declare_parameter("max_accel", 2.0);         // 0.11G
+  p.min_decel = declare_parameter("min_decel", -3.0);        // -0.2G
```

Comment:
I'd use a macro to avoid repeating the parameter name twice as I did here https://github.com/tier4/AutowareArchitectureProposal/pull/101/files#diff-055ef39b3e30ae660ffb81cbd3606f63b0a801dde224b2893ab9fc229db79fe8R231

Reply:
done